compose/src/main/java/com/bottlerocketstudios/compose/utils/map/MapClusterIconUtils.kt

```diff
@@ -0,0 +1,94 @@
+package com.bottlerocketstudios.compose.utils.map
+
+import android.content.Context
+import android.graphics.Color
+import android.graphics.drawable.Drawable
+import android.graphics.drawable.LayerDrawable
+import android.graphics.drawable.ShapeDrawable
+import android.graphics.drawable.shapes.OvalShape
+import android.util.SparseArray
+import android.view.ViewGroup
+import androidx.core.util.isNotEmpty
+import com.bottlerocketstudios.compose.R
+import com.google.android.gms.maps.model.BitmapDescriptor
+import com.google.android.gms.maps.model.BitmapDescriptorFactory
+import com.google.maps.android.clustering.Cluster
+import com.google.maps.android.ui.IconGenerator
+import kotlin.math.pow
+
+// This is used to determine which bucket each cluster would go in.
+private val buckets = intArrayOf(10, 20, 50, 100, 200, 500, 1000)
+private val icons = SparseArray<BitmapDescriptor>()
+private var circleBackground: ShapeDrawable? = null
+private const val DEFAULT_SATURATION = 1f
+private const val DEFAULT_VALUE = .6f
+private const val HUE_RANGE = 220f
+private const val SIZE_RANGE = 300f
+private const val PADDING = 12
+private const val INDEX_ONE = 1
+private const val TRANSPARENT_WHITE = -0x7f000001
+private const val STROKE_MODIFIER = 3
+fun clusterIconGenerator(context: Context, cluster: Cluster<MapClusterItem>): BitmapDescriptor {
```

Comment:
@Josh-Gebbeken If this implementation is based off of or inspired by another one, it would be nice to add a link to this function so that any future updates/changes could be reviewed and potentially integrated

Reply:
This has been changed. Not sure why this doesn't say Outdated.